packages/xo-server-test/src/_xoConnection.js

```diff
@@ -155,6 +162,20 @@ class XoConnection extends Xo {
     })
   }
 
+  async cloneTempVm(id) {
+    const clonedVmId = await this.call('vm.clone', {
+      full_copy: false,
+      id,
+      name: getDefaultName(),
+    })
+    this._tempResourceDisposers.push('vm.delete', { id: clonedVmId })
+    return this.waitObjectState(id, vm => {
```

Comment:
Why not update `waitObjectState` to not call the predicate if the object is not there yet?

Example: https://github.com/vatesfr/xen-orchestra/blob/26a295c8ed5fce313b735ec58d0c314ad29be3da/packages/xo-server/src/xapi/index.js#L229-L230

Reply:
Did you address this?

Reply:
i used `getOrWaitObject` instead, it's more pertinent.